cpp/src/DataStorm/NodeI.cpp

```diff
@@ -246,15 +271,32 @@ NodeI::confirmCreateSession(
     checkNotNull(publisherSession, __FILE__, __LINE__, current);
 
     unique_lock<mutex> lock(_mutex);
+    auto instance = _instance.lock();
+    assert(instance);
```

Comment:
Would be good to add a comment on why it's always not null.

Reply:
Added a comment here, and in other similar places.